website/content/docs/internals/integrated-storage.mdx

```diff
@@ -1,321 +1,279 @@
+before they can commit a log entry. If one of the nodes fails, the remaining
+node cannot reach quorum. The peer set can no longer add or remove nodes or
+commit additional log entries. When the peer set can no longer take action, it
+becomes **unavailable**. Once a peer set becomes unavailable, it can only be
+recovered manually by removing the failing node and restarting the remaining
+node in bootstrap mode so it self-elects as leader.
+
+## Raft leadership in Vault
+
+When a single Vault server (node)
+[initializes](/vault/docs/commands/operator/init/#operator-init), it establishes
+a cluster (peer set) of size 1 and self-elects itself as leader. Once the
+cluster has a leader, additional servers can join the cluster using an 
+encrypted challenge/answer workflow. For the join process to work, all all nodes
+in a single Raft cluster must share the same seal configuration. If the cluster
+is configured to use auto-unseal, the join process automatically decrypts the
+challenge and responds with the answer using the configured seal. For other seal
+options, like a Shamir seal, nodes must have access to the unseal keys before
+joining so they can decrypt the challenge and respond with the decrypted answer.
+
+In a [high availability](/vault/docs/internals/high-availability#design-overview)
+configuration, the active Vault node is the leader node and all standby nodes
+are followers.
+
+## BoltDB Raft logs
+
+BoltDB is a single file database, which means BoltDB cannot shrink the file on
+disk to recover space when you delete data. Instead, BoltDB notes the places
+where the deleted data was stored on a "freelist". On subsequent writes, BoltDB
+consults the freelist to reuse old pages before allocating new space to persist
+the data.
+
+<Warning title="BoltDB requires careful tuning">
+
+1. On Vault clusters with high churn, the BoltDB freelist can become quite large
+   and the database file can become highly fragmented. Large freelists and
+   fragmented database files can slow BoltDB transaction and directly impact the
+   performance of your Vault cluster.
+1. On busy Vault clusters, where new followers struggle to sync Raft snapshots
+   before receiving subsequent snapshots from the leader, the BoltDB file is
+   susceptible to sudden bursts of writes. Not only will new followers potentially
+   fail to join quorum, Vault installations that do not provide for spiky file
+   growth or over-allocate and waste disk space will likely see poor performance.
+
+</Warning>
+
+## Write-ahead Raft logs
+
+@include 'alerts/experimental.mdx'
+
+By default, Vault uses the `raft-boltdb` library for BoltDB to store Raft logs,
+but you can also configure Vault to use the
+[`raft-wal`](https://github.com/hashicorp/raft-wal) library for write-ahead Raft
+logs.
+
+Library       | Filename  | Storage directory
+------------- | --------- | ----------------
+`raft-boltdb` | `raft.db` | `raft`
+`raft-wal`    | `raft.db` | `raft/wal`
```

Comment:
The filename for `raft-wal` is not `raft.db`. `raft-wal` writes many different files as part of its operation.